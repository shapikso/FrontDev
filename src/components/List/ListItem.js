import React from 'react';
import {ButtonWrapper, StList} from "./styled";
import {StButton} from "../commonStyles/commonStyles";

const ListItem = ({title, checked, changeChecked, id, deleteTodo}) => {
    const deleteTask = () => deleteTodo(id);
    const changeTask = () => changeChecked(id);

    return(
        <StList done={checked}>
            <p>{title}</p>
            <ButtonWrapper>
                <StButton width="40px" onClick={ deleteTask }><i className="fas fa-trash"></i></StButton>
                <StButton width="40px" onClick={ changeTask }><i className="fas fa-check"></i></StButton>
            </ButtonWrapper>
        </StList>
    );};

export default ListItem;