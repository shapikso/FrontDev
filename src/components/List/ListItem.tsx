import React from 'react';
import {ButtonWrapper, StList} from "./styled";
import {StButton} from "../commonStyles/commonStyles";
import {TTheme} from "../../store/theme/types";

type TProps = {
    title : string,
    completed: boolean,
    id: number,
    checkTodo: (id: number) => void,
    deleteTodoAction: (id: number) => void,
    showNotification: (type: string, message: string) => void,
    theme: TTheme
}

const ListItem = ({title, completed, id, checkTodo, deleteTodoAction, showNotification, theme }: TProps) => {
    const changeChecked = () => {
        checkTodo(id);
        showNotification('success','Changed Successfully');
    };
    const deleteTodo = () =>{
        deleteTodoAction(id);
        showNotification('success','Delete Successfully');
    };

    return(
        <StList done={completed}>
            <p>{title}</p>
            <ButtonWrapper>
                <StButton theme={theme} width="40px" onClick={ deleteTodo }><i className="fas fa-trash"/></StButton>
                <StButton theme={theme} width="40px" onClick={ changeChecked }><i className="fas fa-check"/></StButton>
            </ButtonWrapper>
        </StList>
    );};

export default ListItem;
