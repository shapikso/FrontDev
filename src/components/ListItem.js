import React from 'react';

const ListItem = ({taskText, checked, changeChecked, id, delTodo}) => (
    <div className={checked ? 'list done' : 'list'}>
        <p>{taskText}</p>
        <div className="divBtn">
            <button onClick={() => delTodo(id)}><i className="fas fa-trash"></i></button>
            <button onClick={() => changeChecked(id)}><i className="fas fa-check"></i></button>
        </div>
    </div>
);

export default ListItem;