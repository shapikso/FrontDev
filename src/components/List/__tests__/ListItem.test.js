import React from 'react';
import {shallow, mount} from 'enzyme';
import ListItem from "../ListItem";
describe('ListItem', ()=> {

    const props = {
        value: 'ToDo testing text' ,
        checked: false,
        changeChecked: jest.fn(),
        id: 20,
        deleteTodo: jest.fn()
    };

    it('should render correctly', () => {
        const component = shallow(<ListItem {...props} />);
        expect(component).toMatchSnapshot();
    });

    it('should render checked prop', () => {
        const component = mount(<ListItem {...props} />);
        expect(component.props().checked).toEqual(props.checked);
    });
    it('should render value', () => {
        const component = mount(<ListItem {...props} />);
        expect(component.find('p').text()).toEqual(props.value);
    });

    it('should render deleteTodo prop', () => {
        const component = mount(<ListItem {...props} />);
        component.find('button').at(0).getElement().props.onClick();
        expect(props.deleteTodo).toHaveBeenCalledWith(props.id);
    });
    it('should render changeChecked prop', () => {
        const component = mount(<ListItem {...props} />);
        component.find('button').at(1).getElement().props.onClick();
        expect(props.changeChecked).toHaveBeenCalledWith(props.id);
    });

});

