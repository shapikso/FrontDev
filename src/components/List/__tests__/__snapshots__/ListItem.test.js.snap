// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem It should render correctly 1`] = `ShallowWrapper {}`;

exports[`ListItem should render correctly 1`] = `ShallowWrapper {}`;
